### PR TITLE
NMS-14154: Add OIA doc

### PR DESCRIPTION
**Pull request by cgorantla**
### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14154

### CODE REVIEW

File: docs/modules/development/pages/oia/introduction.adoc [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+** Graph API extensions (i.e., custom topology)
+
+== Start and build a new project with OIA
+
+OIA provides a Maven archetype to help bootstrap plugin creation.
+
+Create a new plugin project using below maven command :
+
+```
+mvn archetype:generate -B -DarchetypeGroupId=org.opennms.integration.api -DarchetypeArtifactId=example-kar-plugin -DarchetypeVersion=1.0.0-SNAPSHOT -DgroupId=com.company -DartifactId=myplugin -Dversion=0.1.0-SNAPSHOT -Dpackage=com.company.myplugin -DpluginId=myplugin -DpluginName="My Plugin"
```
Comment (cgorantla) [inline docs/modules/development/pages/oia/introduction.adoc:53]: May have to update `archetypeVersion` once  OIA gets released

File: docs/modules/development/pages/oia/introduction.adoc [hunk 2]
```diff
@@ -43,6 +43,34 @@ See the interfaces defined in the `api` module for a complete list.
 ** https://github.com/OpenNMS/opennms-integration-api/blob/master/api/src/main/java/org/opennms/integration/api/v1/timeseries/README.md[Time series storage]
 ** Graph API extensions (i.e., custom topology)
 
-== Start a new project with OIA
+== Start and build a new project with OIA
 
-== Build a new plugin
+OIA provides a Maven archetype to help bootstrap plugin creation.
+
+Create a new plugin project using below maven command :
```
Comment (Bonrob2) [inline docs/modules/development/pages/oia/introduction.adoc:50]: Create a new plugin project using the following Maven command:

File: docs/modules/development/pages/oia/introduction.adoc [hunk 3]
```diff
@@ -43,6 +43,34 @@ See the interfaces defined in the `api` module for a complete list.
...
+
+```
+mvn archetype:generate -B -DarchetypeGroupId=org.opennms.integration.api -DarchetypeArtifactId=example-kar-plugin -DarchetypeVersion=1.0.0-SNAPSHOT -DgroupId=com.company -DartifactId=myplugin -Dversion=0.1.0-SNAPSHOT -Dpackage=com.company.myplugin -DpluginId=myplugin -DpluginName="My Plugin"
+```
+
+NOTE: You can replace name `myplugin` and package name `com.company.myplugin` and version with your own choices. Generated plugin adds some Java files and feature definitions which are only added as sample. They can be updated and deleted when needed.
```
Comment (Bonrob2) [inline docs/modules/development/pages/oia/introduction.adoc:56]: NOTE: You can replace name `myplugin`, package name `com.company.myplugin`, and version with your own choices. 
Generated plugin adds some Java files and feature definitions which are only added as samples. 
You can update and delete them when needed.

File: docs/modules/development/pages/oia/introduction.adoc [hunk 4]
```diff
@@ -43,6 +43,34 @@ See the interfaces defined in the `api` module for a complete list.
...
+
+This plugin generates the following folder structure :
```
Comment (Bonrob2) [inline docs/modules/development/pages/oia/introduction.adoc:58]: This plugin generates the following folder structure:

File: docs/modules/development/pages/oia/introduction.adoc [hunk 5]
```diff
@@ -43,6 +43,34 @@ See the interfaces defined in the `api` module for a complete list.
...
+assembly  <2>
+karaf-features <3>
+plugin   <4>
+pom.xml  <5>
+----
+
+<1> Readme file on how to build and deploy plugin on {page-component-title}.
+<2> assembly builds `kar` file that can be deployed in {page-component-title}.
+<3> karaf-features consists of feature definitions.
+<4> plugin code that depends on OIA code.
```
Comment (Bonrob2) [inline docs/modules/development/pages/oia/introduction.adoc:71]: <4>Plugin code that depends on OIA code.

**Review by Bonrob2 — CHANGES_REQUESTED** [review]

File: docs/modules/development/pages/oia/introduction.adoc [hunk 6]
```diff
@@ -43,6 +43,34 @@ See the interfaces defined in the `api` module for a complete list.
 ** https://github.com/OpenNMS/opennms-integration-api/blob/master/api/src/main/java/org/opennms/integration/api/v1/timeseries/README.md[Time series storage]
 ** Graph API extensions (i.e., custom topology)
 
-== Start a new project with OIA
+== Start and build a new project with OIA
```
Comment (Bonrob2) [inline docs/modules/development/pages/oia/introduction.adoc:46]: Start to build a new project with OIA

Comment (Bonrob2) [reply docs/modules/development/pages/oia/introduction.adoc]: @cgorantla could you add this one as well. Everything else looks good.

**Comment by cgorantla** [comment]
@Bonrob2 Thanks for the review. I took care of all of suggested changes.

File: docs/modules/development/pages/oia/introduction.adoc (hunk 6 above)
Comment (Bonrob2) [reply docs/modules/development/pages/oia/introduction.adoc:46]: I think I understand why this was changed back to "start and build".

**Review by Bonrob2 — APPROVED** [review]

**Pull request merged**